node/client/src/lib.rs

```diff
@@ -281,6 +284,19 @@ impl Client {
         (part_add, part_pending)
     }
 
+    /// Checks that the cached hash matches the content of the block
+    pub fn verify_block_hash(beacon_block: &SignedBeaconBlock, shard_block: &SignedShardBlock) -> bool {
+        shard_block.hash == hash_struct(&shard_block.body.header) &&
+            beacon_block.hash == hash_struct(&beacon_block.body.header) &&
+            beacon_block.body.header.shard_block_hash == shard_block.hash
+    }
+
+    /// Gets BLS keys for validating GroupSignature at block_index
+    pub fn get_authority_keys(&self, block_index: u64) -> Vec<BlsPublicKey> {
+        let (_, authority_map) = self.get_uid_to_authority_map(block_index);
```

Comment:
off topic here: but @nearmax why is it a authority *map*? it has ids from 0 to num_authorities. Why is not just a Vec?

Reply:
Historical reasons from the times of continuous TxFlow and rotating authorities. Currently we are also planning on rotating authorities, but this is done in a more clean-cut manner where at each block we have a different group of authorities. While in continuous TxFlow an each group is associated with some epoch and these groups can overlap in time, meaning one group is still exiting the DAG, while another one is already building in exactly the same DAG. I do not remember exactly with Vec representation but it was problematic to express stuff like this with Vec only.